listenbrainz/webserver/views/stats_api.py

```diff
@@ -550,3 +584,29 @@ def _get_entity_list(
     elif entity == 'recording':
         return getattr(stats, stats_range).recordings[offset:count]
     raise APIBadRequest("Unknown entity: %s" % entity)
+
+
+def _get_country_codes_by_msids(artist_msids):
+    country_map = defaultdict(int)
+    for index in range(0, len(artist_msids), 50):
+        current_batch = artist_msids[index: index + 50]
+        try:
+            artist_credit_id_data = requests.get('http://bono.metabrainz.org:8000/artist-msid-lookup/json', {
+                '[artist_msid]': ','.join(current_batch)
+            }).json()
+            artist_credit_ids = [
+                x['artist_credit_id'] for x in artist_credit_id_data]
+
+            location_data = requests.get('http://bono.metabrainz.org:8000/artist-credit-id-country-code/json', {
```

Comment:
The dataset isn't returning country codes for a majority of artists. we need to look into that.